### PR TITLE
Remove "none" logging level

### PR DESCRIPTION
**Pull request by nick-cd**
This PR removes the NONE/None logging level as those enum constants/variants are never used by the client code.

### CODE REVIEW

**Comment by Minoru** [comment]
I restarted FreeBSD 11.4 task because it failed to build gettext. This happens sometimes, I haven't looked into it yet. Definitely unrelated to your changes.

**Comment by nick-cd** [comment]
I completed the task, but I feel that some changes that I made are not entirely correct especially with the `rs_cliargsparser_log_level()` function.

**Review by Minoru — CHANGES_REQUESTED** [review]
This actually looks great. The `rs_cliargsparser_log_level()` is completely correct as far as I can see ;) Most of my comments are actually a fallout from previous docs being wrong. I'm the one who wrote them, so sorry about that!

Apart from the comments on the specific lines, I have a more general concern: we no longer have tests for "none" level. But that use-case is pretty important, since it's exactly how Newsboat runs most of the time: with no logging configured. @nick-cd, do you see any way to bring those tests back somehow?

Thanks for working on this!

**Comment by nick-cd** [comment]
Before I make any changes, I want to see what you think of my answer to your comment:

> Apart from the comments on the specific lines, I have a more general concern: we no longer have tests for "none" level. But that use-case is pretty important, since it's exactly how Newsboat runs most of the time: with no logging configured. @nick-cd, do you see any way to bring those tests back somehow?

This is what I was thinking (written as a TODO list):

- Modify the `levels` data member in the LogLinesCounter struct to be: `Vec<Option<Level>>` (Originally `Vec<Level>`). 
- Replace `Level::None` with `None` where necessary
- In the `test()` function (`test(&self) -> io::Result<()>`), I can replace the call to `set_loglevel()`:
```rust
for level in &self.levels {
    //...
    logger.set_loglevel(*level);
    //...
}
```
to this:
```rust
for level in &self.levels {
    //...
    match *level {
        Some(l) => logger.set_loglevel(l),
        None => logger.unset_loglevel()
    };
    //...
}
```

File: include/logger.h [hunk 1]
```diff
@@ -9,25 +9,25 @@ enum class Level;
...
 void set_user_error_logfile(const std::string& logfile);
 void set_loglevel(Level l);
+void unset_loglevel();
 
 template<typename... Args>
 void log(Level l, const std::string& format, Args... args)
 {
-	if (l == Level::USERERROR
-		|| static_cast<uint64_t>(l) <= rs_get_loglevel()) {
+	if (static_cast<int64_t>(l) <= rs_get_loglevel()) {
```
[the remark this answers is not in the document]
Comment (nick-cd) [reply include/logger.h:30]: > I wrote that, so please take my apology for misleading you

No worries, really though, I should have re-read the docs before I made such a change :zipper_mouth_face: 

> The l == Level::USERERROR thing is intentional: if the user set error-log, all USERERROR messages are written to that file, even when no --log-level option is passed to Newsboat. The docstring for set_user_error_logfile had this wrong

Now, Just to be sure I understand, 

Regardless of the value of loglevel, so long as the user has set `error-log` in their configuration file, `Level::USERERROR` messages will _always_ be written.

Is that right?

**Comment by Minoru** [comment]
@nick-cd Sorry for the delay! Yes, what you describe sounds like a perfect fix :+1:

File: include/logger.h (hunk 1 above)
[the remark this answers is not in the document]
Comment (Minoru) [reply include/logger.h:30]: Yes, that's right (modulo technicalities — error-log couldn't be opened, the disk space ran out, the operating system failed to write the stuff onto disk, etc.).

**Review by Minoru — APPROVED** [review]
Sorry for the delay, @nick-cd ­— GitHub doesn't email me when a PR is marked ready for review :(

These changes look excellent, so I'm ready to merge this! You probably want to squash your fixes, please drop a comment once you've done that.

Thank you for the pull request! I'm glad to see you're taking the more and more complicated tasks, it feels nice to work in a team :)

**Comment by nick-cd** [comment]
> Sorry for the delay, @nick-cd ­— GitHub doesn't email me when a PR is marked ready for review :(

oops! I will be sure to mention you next time.

> These changes look excellent, so I'm ready to merge this! You probably want to squash your fixes, please drop a comment once you've done that.

Done!

> Thank you for the pull request! I'm glad to see you're taking the more and more complicated tasks, it feels nice to work in a team :)

Definitely, I'm happy to help! :smile:
